server/sonar-db-dao/src/main/java/org/sonar/db/component/ComponentDto.java

```diff
@@ -221,6 +226,18 @@ public ComponentDto setProjectUuid(String projectUuid) {
     return this;
   }
 
+  /**
+   * Return the root project key. On a root project, return itself
+   */
+  public String projectKey() {
```

Comment:
Description is wrong : this method only returns something when selectByQuery is used.......